### PR TITLE
feat(dashboard): Nv 4479 workflow editor configure workflow drawer

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What changed? Why was the change needed?

- tooltips
- copy button
- configure workflow section
- Tags show/hide functionality
- fix: formschema
- success alpha colors vars
### Screenshots

https://github.com/user-attachments/assets/2c329500-bd90-4e6d-9f0c-45c6571564f8

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/primitives/copy-button.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <Tooltip>
+        <TooltipTrigger asChild>
+          <Button
+            variant="outline"
+            size="icon"
+            className={className}
+            onClick={copyToClipboard}
+            aria-label="Copy to clipboard"
+          >
+            {isCopied ? <RiCheckLine className="h-4 w-4" /> : <RiFileCopyLine className="h-4 w-4" />}
```
Comment (LetItRock) [inline apps/dashboard/src/components/primitives/copy-button.tsx]: Is this what we agreed on with Naveen?

File: apps/dashboard/tailwind.config.js [hunk 2]
```diff
@@ -72,6 +72,19 @@ export default {
...
+        200: 'hsl(var(--success-alpha-200))',
+        300: 'hsl(var(--success-alpha-300))',
+        400: 'hsl(var(--success-alpha-400))',
+        500: 'hsl(var(--success-alpha-500))',
+        600: 'hsl(var(--success-alpha-600))',
+        700: 'hsl(var(--success-alpha-700))',
+        800: 'hsl(var(--success-alpha-800))',
+        900: 'hsl(var(--success-alpha-900))',
+        950: 'hsl(var(--success-alpha-950))',
+      },
```
Comment (LetItRock) [inline apps/dashboard/tailwind.config.js]: there is no need for `success-alpha`, as you can do `bg-success/20`

File: apps/dashboard/src/pages/edit-workflow.tsx [hunk 3]
```diff
@@ -12,30 +9,36 @@ import {
...
+import { ConfigureWorkflow } from '@/components/workflow-editor/configure-workflow';
+import { formSchema } from '@/components/workflow-editor/schema';
 import { useEnvironment } from '@/context/environment/hooks';
-import { useFetchWorkflow } from '@/hooks/use-fetch-workflow';
 import { buildRoute, ROUTES } from '@/utils/routes';
-import { Toaster } from '@/components/primitives/sonner';
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import * as z from 'zod';
```
Comment (LetItRock) [inline apps/dashboard/src/pages/edit-workflow.tsx]: please move these imports to the top

File: apps/dashboard/src/components/workflow-editor/configure-workflow.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          )}
+        />
+        <FormField
+          control={control}
+          name="description"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Description</FormLabel>
+              <FormControl>
+                <Textarea placeholder="Untitled" {...field} />
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/configure-workflow.tsx]: placeholder: `Description of what this workflow does`

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 5]
```diff
@@ -52,21 +79,37 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
...
-          value={inputValue}
-          onKeyDown={handleKeyDown}
-          className={cn('flex-grow', className)}
-          placeholder="Type a tag and press Enter"
-          onChange={(e) => setInputValue(e.target.value)}
-          {...rest}
-        />
-      </InputField>
+    <div className="flex flex-col space-y-2" ref={divRef}>
+      {showAddButton && isFocussed && (
```
Comment (LetItRock) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: @BiswaViraj, please speak to @desiprisg about this field, as there were some changes to how it should work 

---
the focused state could be handled with css and we can animate nicely the add button

File: apps/dashboard/src/components/workflow-editor/configure-workflow.tsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+          <FormItem className="flex items-center justify-between gap-2.5 space-y-0 px-3 py-2">
+            <div className="flex items-center gap-4">
+              <div
+                className="bg-success-alpha-600 data-[active=false]:shadow-neutral-alpha-100 ml-2 h-1.5 w-1.5 rounded-full [--pulse-color:var(--success)] data-[active=true]:animate-[pulse-shadow_1s_ease-in-out_infinite] data-[active=false]:bg-neutral-300 data-[active=false]:shadow-[0_0px_0px_5px_var(--neutral-alpha-200),0_0px_0px_9px_var(--neutral-alpha-100)]"
+                data-active={field.value}
+              />
+              <FormLabel>Active Workflow</FormLabel>
+            </div>
+            <FormControl>
+              <Switch checked={field.value} onCheckedChange={field.onChange} />
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/configure-workflow.tsx:40]: the toggle size looks too big from what we have in the designs
Screenshot 2024-10-22 at 00 29 28

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 7]
```diff
@@ -79,10 +122,18 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
             </button>
           </Badge>
         ))}
+
+        {showAddButton && !isFocussed && (
+          <Badge variant="outline" kind="tag">
+            <button type="button" className="rounded-full outline-none focus:outline-none">
```
Comment (LetItRock) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: please update the button size as it looks wider than in the designs
Screenshot 2024-10-22 at 00 29 28

File: apps/dashboard/src/components/workflow-editor/configure-workflow.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+                  </TooltipContent>
+                </Tooltip>
+              </TooltipProvider>
+            </div>
+            <FormControl>
+              <Switch checked={field.value} onCheckedChange={field.onChange} />
+            </FormControl>
+          </FormItem>
+        )}
+      />
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/configure-workflow.tsx]: There have been some changes in the designs around the preferences. Let's remove this field and add the new "Configure preferences" button (but let's keep the schema), and we will do that in a separate ticket. Please add the placeholder ticket and ask me to fill in the rest.

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 9]
```diff
@@ -52,21 +79,37 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
...
+      {showAddButton && isFocussed && (
+        <InputField>
+          <Input
+            ref={ref}
+            type="text"
+            value={inputValue}
+            onKeyDown={handleKeyDown}
+            className={cn('flex-grow', className)}
+            placeholder="Type a tag and press Enter"
+            onChange={(e) => setInputValue(e.target.value)}
```
Comment (LetItRock) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: I noticed that somehow the font-size and placeholder font-size are not aligned with the rest of inputs
Screenshot 2024-10-22 at 00 38 17

**Review by LetItRock — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 10]
```diff
@@ -3,18 +3,45 @@
...
   onChange: (tags: string[]) => void;
+  showAddButton?: boolean;
 };
 
 const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
-  const { className, value, onChange, ...rest } = props;
+  const { className, value, onChange, showAddButton, ...rest } = props;
   const [tags, setTags] = useState<string[]>(value);
   const [inputValue, setInputValue] = useState('');
+  const [isFocussed, setIsFocussed] = useState(false);
```
Comment (desiprisg) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: `isFocused` :D

File: apps/dashboard/src/index.css [hunk 11]
```diff
@@ -55,6 +55,17 @@
...
+    --success-alpha-100: 148 72% 44% / 0.1;
+    --success-alpha-200: 148 72% 44% / 0.2;
+    --success-alpha-300: 148 72% 44% / 0.3;
+    --success-alpha-400: 148 72% 44% / 0.4;
+    --success-alpha-500: 148 72% 44% / 0.5;
+    --success-alpha-600: 148 72% 44% / 0.6;
+    --success-alpha-700: 148 72% 44% / 0.7;
+    --success-alpha-800: 148 72% 44% / 0.8;
+    --success-alpha-900: 148 72% 44% / 0.9;
+    --success-alpha-950: 148 72% 44% / 0.95;
```
Comment (desiprisg) [inline apps/dashboard/src/index.css]: Why is this needed?

File: apps/dashboard/tailwind.config.js (hunk 2 above)
Comment (desiprisg) [reply apps/dashboard/tailwind.config.js]: +1

**Review by LetItRock — APPROVED** [review]
🙌
